shoop_tests/functional/test_discount_taxing.py

```diff
@@ -0,0 +1,544 @@
+from collections import defaultdict
```

Comment:
Add ``from __future__ import unicode_literals`` for python 2 also you might want to add the license header.